std-bits/table/src/main/java/org/enso/table/excel/xssfreader/XSSFReaderWorkbook.java

```diff
@@ -0,0 +1,299 @@
+    if (readShared) {
+      return;
+    }
+
+    try {
+      withReader(
+          rdr -> {
+            try {
+              rdr.setUseReadOnlySharedStringsTable(true);
+              sharedStrings = rdr.getSharedStringsTable();
+              if (sharedStrings == null) {
+                sharedStrings =
+                    new SharedStrings() {
+                      @Override
+                      public RichTextString getItemAt(int idx) {
+                        return null;
+                      }
+
+                      @Override
+                      public int getCount() {
+                        return 0;
+                      }
+
+                      @Override
+                      public int getUniqueCount() {
+                        return 0;
+                      }
+                    };
+              }
+
+              // Read the styles table and attach the format data
+              var stylesTable = rdr.getStylesTable();
+              styles = new XSSFReaderFormats(stylesTable);
+
+              readShared = true;
+            } catch (InvalidFormatException | IOException e) {
+              throw new RuntimeException(e);
+            }
+          });
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /** Flag that workbook is in 1904 format. */
+  boolean use1904Format() {
+    return use1904DateSystemFlag;
+  }
+
+  private Map<String, SheetInfo> getSheetInfoMap() {
+    return sheetInfoMap;
+  }
+
+  private List<SheetInfo> getSheetInfos() {
+    return sheetInfos;
+  }
+
+  private Map<String, NamedRange> getNamesMap() {
+    return namedRangeMap;
+  }
+
+  @Override
+  public int getNumberOfSheets() {
+    return getSheetInfoMap().size();
+  }
+
+  @Override
+  public int getSheetIndex(String name) {
+    var map = getSheetInfoMap();
+    if (!map.containsKey(name)) {
+      return -1;
+    }
+    return map.get(name).index;
+  }
+
+  @Override
+  public String getSheetName(int sheet) {
+    var list = getSheetInfos();
+    if (sheet < 0 || sheet >= list.size()) {
+      throw new IllegalArgumentException("Sheet index out of range: " + sheet);
+    }
+    return list.get(sheet).name;
+  }
+
+  @Override
+  public int getNumberOfNames() {
+    return getNamesMap().size();
+  }
+
+  @Override
+  public String[] getRangeNames() {
+    return getNamesMap().keySet().toArray(String[]::new);
+  }
+
+  @Override
+  public String getNameFormula(String name) {
+    var map = getNamesMap();
+    var namedRange = map.get(name);
+    return namedRange == null ? null : namedRange.formula;
+  }
+
+  public SharedStrings getSharedStrings() {
+    if (!readShared) {
+      readShared();
+    }
+    return sharedStrings;
+  }
+
+  public XSSFReaderFormats getStyles() {
+    if (!readShared) {
+      readShared();
+    }
+    return styles;
+  }
+
+  @Override
+  public ExcelSheet getSheetAt(int sheetIndex) {
+    var sheetInfos = getSheetInfos();
```

Comment:
This local variable is shadowing a member field with the same name. I'd suggest to rename it as it is very easy recipe for confusion.

I was also surprised to see the `getSheetInfos()` calls everywhere even if all that getter does is just return a member field. I assume it is because in some earlier iterations these were loaded lazily? I guess it is okay to keep it that way in case we want to bring back lazy loading (although I imagine it was removed because it wasn't really beneficial?)